### PR TITLE
Adding build scripts

### PR DESCRIPTION
**Pull request by aequasi**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,9 +30,8 @@ script:
   - vendor/bin/phpunit -c phpunit.xml.dist
 
 after_success:
-  - codecov # If coverage
-  - if [ "$TRAVIS_BRANCH" = "master" && "$SUBSPLIT" = true ]; then ''; fi
-  - if [[ "$TRAVIS_BRANCH" = "master" && "$SUBSPLIT" = true ]]; then ''; fi;
+  - if [[ "$SEND_COVERAGE" = true ]]; codecov; fi
+  - if [[ "$TRAVIS_BRANCH" = "master" && "$SUBSPLIT" = true ]]; then ''; fi # Send subtree update
```
Comment (Nyholm) [inline .travis.yml]: This line could be removed

Comment (aequasi) [inline .travis.yml]: It'll just be re-added when you get the heroku thing added

Comment (Nyholm) [inline .travis.yml]: No, we use a webhook instead

File: src/Adapter/Doctrine/composer.json [hunk 2]
```diff
@@ -25,37 +23,32 @@
...
     },
-    "require-dev":
-    {
-        "phpunit/phpunit":         "^5.1|^4.0",
-        "mockery/mockery":         "^0.9.4",
-        "cache/integration-tests": "^0.7"
+    "require-dev":       {
+        "phpunit/phpunit":         "^4.0|^5.1",
+        "cache/integration-tests": "^0.7",
+        "mockery/mockery":         "^0.9.4"
```
Comment (prisis) [inline src/Adapter/Doctrine/composer.json]: no need for the last number

File: src/Adapter/Predis/composer.json [hunk 3]
```diff
@@ -26,32 +24,27 @@
             "homepage": "https://github.com/nyholm"
         }
     ],
-    "require":
-    {
+    "require":           {
         "php":                      "^5.5|^7.0",
-        "psr/cache":                "1.0.0",
+        "psr/cache":                "^1.0",
         "cache/adapter-common":     "^0.1.2",
```
Comment (prisis) [inline src/Adapter/Predis/composer.json]: maybe this?

File: src/Adapter/Void/composer.json [hunk 4]
```diff
@@ -25,30 +23,25 @@
...
-    {
-        "phpunit/phpunit":         "^5.1|^4.0",
+    "require-dev":       {
+        "phpunit/phpunit":         "^4.0|^5.1",
         "cache/integration-tests": "^0.7"
     },
-    "provide":
-    {
+    "provide":           {
         "psr/cache-implementation": "1.0.0"
```
Comment (prisis) [inline src/Adapter/Void/composer.json]: change to ^1.0

File: src/Adapter/Redis/composer.json [hunk 5]
```diff
@@ -26,32 +24,27 @@
...
-    {
-        "phpunit/phpunit":         "^5.1|^4.0",
+    "require-dev":       {
+        "phpunit/phpunit":         "^4.0|^5.1",
         "cache/integration-tests": "^0.7"
     },
-    "provide":
-    {
+    "provide":           {
         "psr/cache-implementation": "1.0.0"
```
Comment (prisis) [inline src/Adapter/Redis/composer.json]: change to ^1.0

File: src/Adapter/Predis/composer.json [hunk 6]
```diff
@@ -26,32 +24,27 @@
...
-        "phpunit/phpunit":         "^5.1|^4.0",
-        "cache/integration-tests": "dev-master"
+    "require-dev":       {
+        "phpunit/phpunit":         "^4.0|^5.1",
+        "cache/integration-tests": "^0.7"
     },
-    "provide":
-    {
+    "provide":           {
         "psr/cache-implementation": "1.0.0"
```
Comment (prisis) [inline src/Adapter/Predis/composer.json]: change to ^1.0

File: src/Adapter/PHPArray/composer.json [hunk 7]
```diff
@@ -25,31 +23,26 @@
...
-    {
-        "phpunit/phpunit":         "^5.1|^4.0",
+    "require-dev":       {
+        "phpunit/phpunit":         "^4.0|^5.1",
         "cache/integration-tests": "^0.7"
     },
-    "provide":
-    {
+    "provide":           {
         "psr/cache-implementation": "~1.0"
```
Comment (prisis) [inline src/Adapter/PHPArray/composer.json]: change to ^1.0

File: src/Adapter/MongoDB/composer.json [hunk 8]
```diff
@@ -25,31 +23,26 @@
...
-        "phpunit/phpunit":         "^5.1|^4.0",
-        "cache/integration-tests": "dev-master"
+    "require-dev":       {
+        "phpunit/phpunit":         "^4.0|^5.1",
+        "cache/integration-tests": "^0.7"
     },
-    "provide":
-    {
+    "provide":           {
         "psr/cache-implementation": "1.0.0"
```
Comment (prisis) [inline src/Adapter/MongoDB/composer.json]: change to ^1.0

File: src/Adapter/Memcached/composer.json [hunk 9]
```diff
@@ -25,32 +23,27 @@
...
-        "phpunit/phpunit":         "^5.1|^4.0",
-        "cache/integration-tests": "^0.6"
+    "require-dev":       {
+        "phpunit/phpunit":         "^4.0|^5.1",
+        "cache/integration-tests": "^0.7"
     },
-    "provide":
-    {
+    "provide":           {
         "psr/cache-implementation": "1.0.0"
```
Comment (prisis) [inline src/Adapter/Memcached/composer.json]: change to ^1.0

**Pull request merged**